### ISSUE TITLE
Map Generation: Increasing city spacing should reduce the spawn rate of map extras

### ISSUE THREAD

**Issue by natsirt721** [CleverRaven/Cataclysm-DDA#44439]
# Is your feature request related to a problem? Please describe.
I like to play the game with high city spacing (7), which leaves a lot of empty land between places of interest. Especially given the reduction in turn time from 6 to 1 second, I find that the default spacing value makes the map feel very small and so I increase the spacing accordingly. However, it seems like the spawn rate for map extras (corpses, military, helo crashes, etc.) is static, and the larger proportions of field and forest on the map make them appear more often. This makes it comparatively easier to find 'rare' spawns like military corpses (with plenty of M4s and 5.56 ammo) which can significantly boost a survivors chances in the first few days, especially if they have a vehicle like a wagon or bike and can cover ground quickly.

tl;dr city size option makes the map feel bigger, but map extra spawn chance not scaling with that makes the extras seem not as rare or interesting as they otherwise might be.

# Describe the solution you'd like
Higher city spacing values will proportionally reduce the map extra spawn chance by approximately the square of the ratio between default (4) and the selected value - a value of 8 would result in the map extra chance being reduce globally by four times. For values lower than 4, a more linear scaling method might be employed to avoid map extras being incredibly common, or even not modifying the map extra chance at all (I don't know how the ratio of map extras feels on low map sizes so I hesitate to postulate what the scaling method should be).

# Describe alternatives you've considered
I don't know much about the mapgen algorithm or how the city spacing option affects things, but I could see other 'rare' map tiles (labs, bunkers, portals, etc.) receiving similar reductions in spawn rate, if that is not also currently implemented.  A worldgen option for modifying the spawn rate of map extras could be nice as well.

Edit: this might be implemented already, and I just got incredibly lucky by finding two soldier corpse extras in quick succession on my first day. In that case, the alternative solution mentioned above for allowing the spawn rate to be customizable still applies.
